tests/integration-tests/tests/ad_integration/test_ad_integration.py

```diff
@@ -155,7 +159,9 @@ def _store_secret(secret):
 
 @pytest.fixture(scope="module")
 def directory_factory(request, cfn_stacks_factory):
-    created_directory_stacks = {}
+    # TODO: do we need to worry about regions?
```

Comment:
Why wouldn't we care about the region?

Reply:
even within the same region clusters won't these tests probably be running in their own processes, and thus not be able to share this data structure?

Reply:
Yes. If the tests are in different processes, they couldn't share the AD.
The idea was to reduce the creation time of AD and share it within each process.
Sharing it across process is not easy (e.g. we need to handle racing condition)

Reply:
yeah agreed. some sort of lock would be needed. the reason i leave the TODO asking if we need to handle regions is because i think we can assume that tests running within the same process will be in the same region. but it doesn't hurt anything to handle it, and it makes it safer. i'll remove the region TODO.